src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -489,3 +489,31 @@ end
                                                                                                                                       })
     return SVector(flux_inner[1], flux_inner[2], flux_inner[3], flux_inner[4])
 end
+
+# Dirichlet Boundary Condition for P4est mesh
+
+@inline function (boundary_condition::BoundaryConditionDirichlet)(flux_inner,
+                                                                  u_inner,
+                                                                  normal::AbstractVector,
+                                                                  x, t,
+                                                                  operator_type::Gradient,
+                                                                  equations::CompressibleNavierStokesDiffusion2D{
+                                                                                                                 GradientVariablesPrimitive
+                                                                                                                 })
+    # BCs are usually specified as conservative variables so we convert it to primitive variables
+    #  because the derivative are calculated with respect to primitive variables
```

Comment:
```suggestion
    # BCs are usually specified as conservative variables so we convert them to primitive variables
    #  because the gradients are assumed to be with respect to the primitive variables
```